cryptarchia/cryptarchia.py

```diff
@@ -37,9 +43,48 @@ def epoch(self, config: TimeConfig) -> Epoch:
 
 
 @dataclass
-class Config:
-    k: int
-    time: TimeConfig
+class Coin:
+    pk: int
+    value: int
+
+    def commitment(self) -> Id:
+        # TODO: mocked until CL is understood
+        pk_bytes = int.to_bytes(self.pk, length=32, byteorder="little")
+        value_bytes = int.to_bytes(self.value, length=32, byteorder="little")
+
+        h = sha256()
+        h.update(pk_bytes)
+        h.update(value_bytes)
+        return h.digest()
+
+    def nullifier(self) -> Id:
+        # TODO: mocked until CL is understood
+        pk_bytes = int.to_bytes(self.pk, length=32, byteorder="little")
+        value_bytes = int.to_bytes(self.value, length=32, byteorder="little")
+
+        h = sha256()
+        h.update(pk_bytes)
+        h.update(value_bytes)
+        h.update(b"\x00")  # extra 0 byte to differentiate from commitment
+        return h.digest()
+
+
+@dataclass
+class MockLeaderProof:
+    commitment: Id
+    nullifier: Id
+
+    @staticmethod
+    def from_coin(coin: Coin):
+        return MockLeaderProof(commitment=coin.commitment(), nullifier=coin.nullifier())
+
+    def verify(self, slot):
+        # TODO: verification not implemented
+        return True
+
+    def _id_update(self, hasher):
```

Comment:
what's this for?

Reply:
It's for the LeaderProof to contribute to the block id without explicitly speccing out this out right now. the details of what a leadership proof will be will depend on the cryptrography we use to implement the leader proof.